docs/design/ha-flow/ha-flow-design.md

```diff
@@ -0,0 +1,76 @@
+# HA-flows
+
+## Concepts / objects model
+Introducing a new Kilda entity - HA-flow. Defined by 3 network endpoints. One "shared" endpoint is defined by
+`switch + port + outer-vlan + inner-vlan`. Two other endpoints are defined by `switch + port + outer-vlan + inner-vlan`.
+We can name these 3 endpoints as Z-end (the shared endpoint), A-end (the first leaf), B-end (the second leaf).
+
+```
+Z-end <===> Y-point <===> A-end
+                    <===> B-end
+```
+
+There are 2 directions in HA-flows: 
+* forward: Z-A and Z-B
+* reverse: A-Z and B-Z
+
+HA-flow delivers all traffic from Z-end to A-end and B-end. It meas that 
+A-end and B-end will receive identical set of packets from Z-end. 
+To reach it OpenFlow group will be installed at Y-point to duplicate forward traffic.
+Reverse traffic (from A-end and B-end) will be joined at Y-point and delivered to Z end. 
+
+For traffic control purpose (bandwidth limit): each endpoint (A, B and Z) has an own meter
+installed to control traffic. Z-end has a single meter to limit both Z-A and Z-B traffic.
+Y-point has a meter to control summary A-end + B-end traffic. 
+It means that if HA-flow has bandwidth 1 Mb/s all segments (Z-Y, Y-A, Y-B) will have 
+same maximum bandwidth 1 Mb/s in both directions.
```

Comment:
How combining the reverse traffic is going to work? Let's assume that Z to A consumes 1Mbpb, the reverse A to Z also 1Mbps, so Y-->Z utilization is 100%. What happens when there are some packets from B to Z?

Reply:
Some packets will be dropped at Y-point in reverse direction. Which packets it will be - depends of luck. 
It can be packets from A-end or from B-end

Reply:
So, is it up to the switch to decide what to drop, there is no any strategy or rule?